File: panel/src/main/java/com/effective/android/panel/interfaces/listener/OnViewClickListener.java
package com.effective.android.panel.interfaces.listener;

import android.view.View;

/**
 * preventing listeners that {@link com.effective.android.panel.PanelSwitchHelper} set these to view from being overwritten
 * Created by yummyLau on 18-7-07
 * Email: yummyl.lau@gmail.com
 * blog: yummylau.com
 */
public interface OnViewClickListener {

    void onClickBefore(View view);
}

File: panel-androidx/src/main/java/com/effective/android/panel/interfaces/listener/OnViewClickListener.kt
package com.effective.android.panel.interfaces.listener

import android.view.View

/**
 * preventing listeners that [com.effective.android.panel.PanelSwitchHelper] set these to view from being overwritten
 * Created by yummyLau on 18-7-07
 * Email: yummyl.lau@gmail.com
 * blog: yummylau.com
 * update 2020/05/08 支持 dsl
 */
interface OnViewClickListener {
    fun onClickBefore(view: View?)
}

private typealias OnClickBefore = (view: View?) -> Unit

class OnViewClickListenerBuilder : OnViewClickListener {

    private var onClickBefore: OnClickBefore? = null

    override fun onClickBefore(view: View?) {
        onClickBefore?.invoke(view)
    }

    fun onClickBefore(onClickBefore: OnClickBefore) {
        this.onClickBefore = onClickBefore
    }
}